pkg/aggregator/sender.go

```diff
@@ -203,32 +205,17 @@ func (s *checkSender) Commit() {
 	s.cyclemetricStats()
 }
 
-func (s *checkSender) GetMetricStats() map[string]int64 {
-	s.priormetricStats.Lock.RLock()
-	defer s.priormetricStats.Lock.RUnlock()
-
-	metricStats := make(map[string]int64)
-	metricStats["MetricSamples"] = s.priormetricStats.MetricSamples
-	metricStats["Events"] = s.priormetricStats.Events
-	metricStats["ServiceChecks"] = s.priormetricStats.ServiceChecks
-	metricStats["HistogramBuckets"] = s.priormetricStats.HistogramBuckets
-
-	return metricStats
+func (s *checkSender) GetSenderStats() (metricStats check.SenderStats) {
+	s.metricStatsLock.RLock()
```

Comment:
Are you sure to lock the correct one?

Reply:
Yes, there's only the one lock for all stats. I simplified it from how it was earlier where there was a lock per stats object. 

I renamed `metricStatsLock` to `statsLock` to hopefully reduce the confusion.

Reply:
Got it, thanks for the clarification.